### PR TITLE
Add general URL-like specifier remapping examples

### PR DESCRIPTION
**Pull request by domenic**
As was noticed in #109, this was not an obvious capability up front.

---

Suggestions for more realistic examples welcome; I'm not super-happy with these.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,6 +192,30 @@ import localeData from "moment/locale/zh-cn.js";
...
+
+```json
+{
+  "imports": {
+    "/app/helpers.mjs": "/app/helpers/index.mjs"
+  }
+}
+```
+
+This remapping ensures that any URL-like imports that resolve to `/app/helpers.mjs`, including e.g. an `import "./helpers.mjs"` from files inside `/app/`, or an `import "../helpers.mjs"` from files inside `/app/models`, will instead resolve to `/app/helpers/index.mjs`. This is probably not a good idea; instead of creating an indirection which obfuscates your code, you should instead just update your source files to import the correct files. But, it is a useful example for demonstrating the capabilities of import maps.
```
Comment (jkrems) [inline README.md]: > This remapping ensures that any URL-like imports that resolve to `/app/helpers.mjs`, including e.g. an `import "./helpers.mjs"` from files inside `/app/`, or an `import "../helpers.mjs"` from files inside `/app/models`, will instead resolve to `/app/helpers/index.mjs`.

Kind of repeating a comment in another thread - this suggests a "partial resolution" step that seems to be missing from the spec so far. It sounds like the idea is:

1. If `url` starts with `./` or `../`, set `url` to `new URL(url, referrer).path`.
2. Apply import mappings to `url`.
3. Assert: `url` isn't a bare specifier.
3. Return `new URL(url, referrer).href`.

Is that about correct?

Comment (domenic) [reply README.md]: More or less. See https://github.com/WICG/import-maps/tree/master/reference-implementation

Comment (jkrems) [reply README.md]: Ah, I missed the fact that the import map gets preprocessed in `parser.js` where `/foo/bar` keys get turned into fully resolved URL keys (if I'm scanning it correctly). Thanks for pointing me in the right direction. :)

**Comment by domenic** [comment]
Going to merge, but would love more realistic examples if someone has ideas.

**Pull request merged**